### PR TITLE
[FSK-16]: Configure Cypress

### PR DESCRIPTION
**Pull request by erfan-goodarzi**
This PR adds cypress configuration.

### CODE REVIEW

**Comment by AmirabbasJ** [comment]
@erfan-goodarzi @ASafaeirad 
I've had to downgrade nodejs because of [this](https://github.com/cypress-io/cypress/issues/22795#issuecomment-1225535244) and [this](https://github.com/cypress-io/cypress/issues/23499#issuecomment-1224214313)

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -48,3 +44,27 @@ jobs:
         uses: chromaui/action@v3
         with:
           projectToken: ${{ secrets.CHROMATIC_PROJECT_TOKEN }}
+
+  cypress-run: 
+    concurrency: ci-${{ github.ref }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
```
Comment (AmirabbasJ) [inline .github/workflows/CI.yml:53]: why not v3?

```suggestion
        uses: actions/checkout@v3
```

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -4,10 +4,6 @@ on:
   pull_request:
     branches:
       - dev
-  push:
-    branches:
-      - main
-      - dev
```
Comment (AmirabbasJ) [inline .github/workflows/CI.yml:10]: why did you delete these lines?

File: package.json [hunk 3]
```diff
@@ -16,7 +16,12 @@
     "coverage": "vitest run --coverage",
     "storybook": "start-storybook -p 6006",
     "build-storybook": "build-storybook",
-    "pr": "./scripts/pr"
+    "pr": "./scripts/pr",
+    "cy:component:run": "cypress run --component",
+    "cy:e2e:run": "cypress run",
+    "cy:open": "cypress open",
+    "test:e2e:run": "start-server-and-test ru http://127.0.0.1:5173 cy:e2e:run",
```
Comment (AmirabbasJ) [inline package.json:23]: is there typo in this line?

**Review by AmirabbasJ — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 4]
```diff
@@ -1,24 +1,28 @@
...
-# Editor directories and files
-.vscode/*
-!.vscode/extensions.json
-.idea
-.DS_Store
-*.suo
-*.ntvs*
-*.njsproj
-*.sln
-*.sw?
```
Comment (ASafaeirad) [inline .gitignore:24]: Seems the line ending is changed here

File: .github/workflows/CI.yml [hunk 5]
```diff
@@ -32,7 +28,7 @@ jobs:
 
       - name: Test
         run: npm run test:ci
-
+
```
[the remark this answers is not in the document]
Comment (AmirabbasJ) [reply .github/workflows/CI.yml:31]: you mean not run it?

[the remark this answers is not in the document]
Comment (ASafaeirad) [reply .github/workflows/CI.yml:31]: That trailing space on line 31.

[the remark this answers is not in the document]
Comment (AmirabbasJ) [reply .github/workflows/CI.yml:31]: I don't know maybe we could have a yaml linter to lint all our yaml files

**Comment by AmirabbasJ** [comment]
@erfan-goodarzi  do a rebase with dev before merging

**Review by AmirabbasJ — APPROVED** [review]

**Pull request merged**